src/sentry/tsdb/snuba.py

```diff
@@ -19,21 +30,84 @@ class SnubaTSDB(BaseTSDB):
     will return empty results for unsupported models.
     """
 
-    # The ``model_columns`` are translations of TSDB models into the required
-    # columns for querying snuba. Keys are ``TSDBModel`` enumeration values,
-    # values are in the form ``(groupby_column, aggregateby_column or None)``.
-    # Only the models that are listed in this mapping are supported.
-    model_columns = {
-        TSDBModel.project: ("project_id", None),
-        TSDBModel.group: ("issue", None),
-        TSDBModel.release: ("tags[sentry:release]", None),
-        TSDBModel.users_affected_by_group: ("issue", "tags[sentry:user]"),
-        TSDBModel.users_affected_by_project: ("project_id", "tags[sentry:user]"),
-        TSDBModel.frequent_environments_by_group: ("issue", "environment"),
-        TSDBModel.frequent_releases_by_group: ("issue", "tags[sentry:release]"),
-        TSDBModel.frequent_issues_by_project: ("project_id", "issue"),
+    # The ``model_query_settings`` and ``model_being_upgraded_query_settings`` are translations of
+    # TSDB models into required settings for querying snuba. Queries for ``model_columns``
+    # directly hit snuba, while queries for ``model_columns_being_upgraded`` hit
+    # redis in the main thread and snuba in a background thread.
+    model_query_settings = {
+        TSDBModel.project: SnubaModelQuerySettings(snuba.Dataset.Events, "project_id", None, None),
+        TSDBModel.group: SnubaModelQuerySettings(snuba.Dataset.Events, "issue", None, None),
+        TSDBModel.release: SnubaModelQuerySettings(
+            snuba.Dataset.Events, "tags[sentry:release]", None, None
+        ),
+        TSDBModel.users_affected_by_group: SnubaModelQuerySettings(
+            snuba.Dataset.Events, "issue", "tags[sentry:user]", None
+        ),
+        TSDBModel.users_affected_by_project: SnubaModelQuerySettings(
+            snuba.Dataset.Events, "project_id", "tags[sentry:user]", None
+        ),
+        TSDBModel.frequent_environments_by_group: SnubaModelQuerySettings(
+            snuba.Dataset.Events, "issue", "environment", None
+        ),
+        TSDBModel.frequent_releases_by_group: SnubaModelQuerySettings(
+            snuba.Dataset.Events, "issue", "tags[sentry:release]", None
+        ),
+        TSDBModel.frequent_issues_by_project: SnubaModelQuerySettings(
+            snuba.Dataset.Events, "project_id", "issue", None
+        ),
     }
 
+    # In getsentry/getsentry:tsdb.py, we check ``model_columns`` to see if a request
+    # should go to snuba. So, for now, for backwards compatibility, alias
+    # ``model_columns`` to ``model_query_settings``.
+    # TODO(manu): use model_query_settings instead of model_columns in getsentry
```

Comment:
We will stop using `model_columns` when https://github.com/getsentry/getsentry/pull/3237/files is merged. Will remove this line then.